qa/rpc-tests/sc_create_2.py

```diff
@@ -0,0 +1,291 @@
+
+        # create with duplicate key in input
+        #------------------------------------
+        cmdInput = FakeDict( [('fee', fee), ('amount', amount), ('amount', 6.0), ('toaddress', str(toaddress))])
+
+        mark_logs("\nNode 1 create SC with duplicate key in input",self.nodes,DEBUG_MODE)
+        try:
+            tx = self.nodes[1].create_sidechain(cmdInput)
+            print "tx=", tx
+            assert_true(False);
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs(errorString,self.nodes,DEBUG_MODE)
+            assert_true("amount" in errorString)
+
+        # create with a missing mandatory key in input
+        #------------------------------------------------
+        cmdInput = {'withdrawalEpochLength': 10, 'toaddress': toaddress, 'scid': "1234", 'fee': fee}
+
+        mark_logs("\nNode 1 create SC with duplicate key in input",self.nodes,DEBUG_MODE)
+        try:
+            tx = self.nodes[1].create_sidechain(cmdInput)
+            print "tx=", tx
+            assert_true(False);
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs(errorString,self.nodes,DEBUG_MODE)
+            assert_true("amount" in errorString)
+
+        # create with a bad value for amount
+        #------------------------------------------------
+        cmdInput = {'withdrawalEpochLength': 10, 'toaddress': toaddress, 'amount': -0.1, 'scid': "1234", 'fee': fee}
+
+        mark_logs("\nNode 1 create SC with an invalid amount in input", self.nodes, DEBUG_MODE)
+        try:
+            tx = self.nodes[1].create_sidechain(cmdInput)
+            print "tx=", tx
+            assert_true(False);
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs(errorString,self.nodes,DEBUG_MODE)
+            assert_true("range" in errorString)
+
+        # create with a bad withdrawal epoch length 
+        #------------------------------------------------
+        cmdInput = {'withdrawalEpochLength': 0, 'toaddress': toaddress, 'amount': 0.1, 'scid': "1234", 'fee': fee}
+
+        mark_logs("\nNode 1 create SC with an invalid epoch length in input", self.nodes, DEBUG_MODE)
+        try:
+            tx = self.nodes[1].create_sidechain(cmdInput)
+            print "tx=", tx
+            assert_true(False);
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs(errorString,self.nodes,DEBUG_MODE)
+            assert_true("withdrawal" in errorString)
+
+        # Node 1 create the SC using a valid input and a fromaddress key/value
+        #------------------------------------------------------------------------------------------
+        mark_logs("\nNode 1 create SC with valid input and a fromaddress key/value", self.nodes,DEBUG_MODE)
+
+        sc_id = "95f5de0829fd3c5e45b63f67beab8c4cb8c1c359ef6e2787aaaf5442d6f4779f"
+        wel = 5
+        fromaddr = []
```

Comment:
Useless, thou shalt not do it, C++ developer!

Reply:
:) ok